python/cuda_parallel/cuda/parallel/experimental/_bindings.pyx

```diff
@@ -0,0 +1,1620 @@
+# distutils: language = c++
+# cython: language_level=3
+# cython: linetrace=True
+
+# Python signatures are declared in the companion Python stub file _bindings.pyi
+# Make sure to update PYI with change to Python API to ensure that Python
+# static type checker tools like mypy green-lights cuda.parallel
+
+from libc.string cimport memset, memcpy
+from libc.stdint cimport uint8_t, uint32_t, uint64_t
+from cpython.bytes cimport PyBytes_FromStringAndSize
+
+from cpython.buffer cimport (
+    Py_buffer, PyBUF_SIMPLE, PyBUF_ANY_CONTIGUOUS,
+    PyBuffer_Release, PyObject_CheckBuffer, PyObject_GetBuffer
+)
+
+import ctypes
+
+cdef extern from "<cuda.h>":
+    cdef struct OpaqueCUstream_st
+    cdef struct OpaqueCUkernel_st
+    cdef struct OpaqueCUlibrary_st
+
+    ctypedef int CUresult
+    ctypedef OpaqueCUstream_st *CUstream
+    ctypedef OpaqueCUkernel_st *CUkernel
+    ctypedef OpaqueCUlibrary_st *CUlibrary
```

Comment:
Could you `cimport` these from `cuda.bindings.cydriver` instead of needing to redefine them? It does add a dependency on `cuda.bindings` but in the future when we migrate `cuda.bindings` to scikit-build-core, I believe we could provide CMake targets that you could link against that would handle linking against the CUDA driver library or other libraries as needed.

Reply:
Again, integration with `cuda.bindings` as well as auto-generation of some of the bindings is deferred to a future PR.